Removing Effect1 parameter and usage. Updated parameter value

SebastianClaesson opened the pull request:
## Overview/Summary

Pull request fixes; 
- https://github.com/Azure/Azure-Landing-Zones/issues/1608


> The Azure Policy Set for "deploy-private-dns-zones" contains a parameter called [effect1](https://github.com/Azure/Enterprise-Scale/blob/b72497573d9d67efc322f5ce147670f9e196b789/src/resources/Microsoft.Authorization/policySetDefinitions/Deploy-Private-DNS-Zones.json#L812-L822) which is used for only these azure policy definitions;
> 
> [DINE-Private-DNS-Azure-EventGridTopics](https://github.com/Azure/Enterprise-Scale/blob/main/src/resources/Microsoft.Authorization/policySetDefinitions/Deploy-Private-DNS-Zones.json#L1436-L1438)
> [DINE-Private-DNS-Azure-IoTHubs](https://github.com/Azure/Enterprise-Scale/blob/b72497573d9d67efc322f5ce147670f9e196b789/src/resources/Microsoft.Authorization/policySetDefinitions/Deploy-Private-DNS-Zones.json#L1478-L1480)
> [DINE-Private-DNS-Azure-EventGridDomains](https://github.com/Azure/Enterprise-Scale/blob/b72497573d9d67efc322f5ce147670f9e196b789/src/resources/Microsoft.Authorization/policySetDefinitions/Deploy-Private-DNS-Zones.json#L1492C11-L1494C12)
> The parameter should be removed and the parameter [effect](https://github.com/Azure/Enterprise-Scale/blob/b72497573d9d67efc322f5ce147670f9e196b789/src/resources/Microsoft.Authorization/policySetDefinitions/Deploy-Private-DNS-Zones.json#L800C6-L811C8) should be used instead across these three policy definitions.

- https://github.com/Azure/Azure-Landing-Zones/issues/209
> However the default value of the parameter dnsZoneNames with key 'azureEventGridTopicsPrivateDnsZoneId' is
>  "privatelink.eventgrid.azure.net"
> Reference; [azureEventGridTopicsPrivateDnsZoneId](https://github.com/Azure/Enterprise-Scale/blob/b72497573d9d67efc322f5ce147670f9e196b789/src/resources/Microsoft.Authorization/policySetDefinitions/Deploy-Private-DNS-Zones.json#L146)
> 
> According to the Event grid documentations the expected DNS zone should be in the following format;
> <TopicName>.<RegionName>.privatelink.eventgrid.azure.net
> 
> Reference; https://learn.microsoft.com/en-us/azure/event-grid/network-security#dns-changes-for-private-endpoints

## This PR fixes/adds/changes/removes

1. Removes Effect1 parameter (#1919)
2. Changes that the three policies that utilized Effect1 now uses Effect parameter (#1919)
3. Updated the parameter for eventgrid topic and domainName to contain regionName (#1920)

### Breaking Changes

1. Removed a parameter called "Effect1" which will lead to that the policySet needs to be reprovisioned.

## Testing Evidence

Using EnterprisePolicyAsCode this is the output; (Proving that a redeploy of the initiative is required)
![image](https://github.com/user-attachments/assets/888dc629-67a0-45c2-9e2b-f71327d16167)
![image](https://github.com/user-attachments/assets/35f14d39-e2c6-4521-8936-3692136c4c2c)

The logic works after update (However I do not have any topics or domain names to test on)
![image](https://github.com/user-attachments/assets/fbb0c5a3-85f6-4540-8beb-38977217f558)


## As part of this Pull Request I have

- [X] Checked for duplicate [Pull Requests](https://github.com/Azure/Enterprise-Scale/pulls)
- [X] Associated it with relevant [issues](https://github.com/Azure/Enterprise-Scale/issues), for tracking and closure.
- [X] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Enterprise-Scale/tree/main)
- [X] Performed testing and provided evidence.
- [X] Ensured [contribution guidance](https://github.com/Azure/Enterprise-Scale/wiki/ALZ-Contribution-Guide) is followed.
- [X] Updated relevant and associated documentation.
- [X] Updated the ["What's New?"](https://github.com/Azure/Enterprise-Scale/wiki/Whats-new) wiki page (located: `/docs/wiki/whats-new.md`)

SebastianClaesson committed:
Removing Effect1 parameter and usage. Replaced with Effect

SebastianClaesson committed:
updating readme

SebastianClaesson committed:
updating with regionName

jtracey93 committed:
feat: add may 2025 community call  (#1989)

Springstone reviewed (changes requested):
@SebastianClaesson many thanks for your contribution and apologies it's taken a while to get to this review. 

Need a couple of changes please:
* Increment initiative version to the next major (as this is a breaking change)
* Move the what's new entry under Policy instead of Tooling
* Rebuild the deployment payload:

`bicep build ./src/templates/initiatives.bicep --outfile ./eslzArm/managementGroupTemplates/policyDefinitions/initiatives.json`

Springstone committed:
Merge branch 'main' into main

leticiavalladares committed:
Update Deny-MgmtPorts-From-Internet.json (#1992)

SebastianClaesson committed:
Merge branch 'Azure:main' into main

SebastianClaesson committed:
Updating to version 3.0.0

SebastianClaesson committed:
Adding note under policy may 2025

SebastianClaesson committed:
Rebuilt the deployment payload

SebastianClaesson commented:
> @SebastianClaesson many thanks for your contribution and apologies it's taken a while to get to this review.
> 
> Need a couple of changes please:
> 
> * Increment initiative version to the next major (as this is a breaking change)
> * Move the what's new entry under Policy instead of Tooling
> * Rebuild the deployment payload:
> 
> `bicep build ./src/templates/initiatives.bicep --outfile ./eslzArm/managementGroupTemplates/policyDefinitions/initiatives.json`

Thanks for the update! :)

- I've updated the version to 3.0.0
- Moved the entry to policy under May 2025.
- Rebuilt the deployment payload

Springstone commented:
@SebastianClaesson many thanks for the changes. I'll redirect the pull request to Policy-Refresh branch, so we can include it with release testing of the next refresh. Really appreciate your contribution!
